src/packages/core/src/servers/ws-server.ts

```diff
@@ -52,7 +52,10 @@ export default class WebsocketServer {
         message: ArrayBuffer,
         isBinary: boolean
       ) => {
-        let payload: ReturnType<typeof connector.parse>;
+        // We have to use type any instead of ReturnType<typeof connector.parse>
+        // on `payload` because Typescript isn't smart enough to understand the
+        // ambiguity doesn't actually exist
+        let payload: any;
```

Comment:
:cry: 

Is the ambiguity caused by just having multiple chains? Or is it specific to the Filecoin API implementation?

Reply:
It's ambiguity caused by having multiple chains, not specific to Filecoin. It's likely that typescript is smart enough with some more convoluted types lol, but I didn't take the appropriate time to learn how to do the complex types yall have set up, because the current types do not support it. I personally don't think it's worth the time for this small functionality